pandas/indexes/multi.py

```diff
@@ -1761,7 +1761,7 @@ def convert_indexer(start, stop, step, indexer=indexer, labels=labels):
 
             else:
                 m = np.zeros(len(labels), dtype=bool)
-                m[np.in1d(labels, r, assume_unique=True)] = True
+                m[np.in1d(labels, r, assume_unique=False)] = True
```

Comment:
if you can gin up examples that are both unique & non-unique just to have them together